### PR TITLE
Internal: Added fonts and colors step [ED-15442]

### PR DESCRIPTION
**Pull request by mugurelLupu**
## PR Checklist

- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: modules/checklist/steps/set-fonts-and-colors.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	const STEP_ID = 'set_fonts_and_colors';
+
+	public function get_id() : string {
+		return self::STEP_ID;
+	}
+
+	public function is_absolute_completed() : bool {
+		$settings = $this->kit_adapter->get_kit_settings();
+		//var_dump($kit);
+		//var_dump($kit->get_settings());
```
Comment (louiswol94) [inline modules/checklist/steps/set-fonts-and-colors.php:22]: Please remove

File: modules/checklist/steps/set-fonts-and-colors.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+		];
+
+		$merged_params = array_merge( $query_params, $additional_params );
+
+		if ( $this->page_exists( $merged_params ) ) {
+			return $this->wordpress_adapter->add_query_arg( $additional_params, $link );
+		}
+
+		return Page::get_elementor_create_new_page_url();
+	}
```
Comment (louiswol94) [inline modules/checklist/steps/set-fonts-and-colors.php:62]: Feels like I've seen this code before. Is it duplicate?

File: modules/checklist/steps/set-fonts-and-colors.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+	public function get_learn_more_url() : string {
+		return 'http://go.elementor.com/app-website-checklist-global-article';
+	}
+
+	public function page_exists( $params ) : bool {
+		$query = new \WP_Query( $params );
+
+		return $query->found_posts;
+	}
```
Comment (louiswol94) [inline modules/checklist/steps/set-fonts-and-colors.php:80]: Duplicate code

File: tests/phpunit/elementor/modules/checklist/classes/step-test-base.php [hunk 4]
```diff
@@ -46,4 +47,8 @@ public function set_wordpress_adapter_mock( $methods, $return_map ) {
 
 		return $wordpress_adapter_mock;
 	}
+
+	public function prophesize ( ?string $classOrInterface = null ): ObjectProphecy{
+		return parent::prophesize($classOrInterface);
+	}
```
Comment (louiswol94) [inline tests/phpunit/elementor/modules/checklist/classes/step-test-base.php:53]: Why do we need this?

**Review by louiswol94 — CHANGES_REQUESTED** [review]

**Comment by ronenelementor** [comment]
Checklist tests are failing (PW)

File: modules/checklist/steps/set-fonts-and-colors.php (hunk 3 above)
Comment (mugurelLupu) [reply modules/checklist/steps/set-fonts-and-colors.php:78]: I removed it from the step where it was not used anymore.

File: tests/phpunit/elementor/modules/checklist/classes/step-test-base.php (hunk 4 above)
Comment (mugurelLupu) [reply tests/phpunit/elementor/modules/checklist/classes/step-test-base.php]: no longer needed, i removed it

File: modules/checklist/steps/set-fonts-and-colors.php (hunk 2 above)
Comment (mugurelLupu) [reply modules/checklist/steps/set-fonts-and-colors.php:60]: No, i removed it from the other step in a previous pr.

Comment (mugurelLupu) [reply modules/checklist/steps/set-fonts-and-colors.php:60]: Not really, is a simillar way of doing things but it is different.

**Pull request closed without merging**